### PR TITLE
Add zsh completion for 'docker {attach,exec,run,start} --detach-keys'

### PR DESCRIPTION
**Pull request by sdurrheimer**
Ref. #15666

Signed-off-by: Steve Durrheimer s.durrheimer@gmail.com

### CODE REVIEW

**Comment by calavera** [comment]
LGTM

**Comment by thaJeztah** [comment]
Thank you! ping @vincentbernat for review :heart:

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -231,6 +231,17 @@ __docker_log_options() {
     return ret
 }
 
+__docker_complete_detach_keys() {
+    [[ $PREFIX = -* ]] && return 1
+    integer ret=1
+
+    compset -P "*,"
+    keys=($(echo {a..z}))
+    ctrl_keys=($(echo ctrl-{{a..z},{@,'[','\\','^',']',_}}))
```
Comment (vincentbernat) [inline contrib/completion/zsh/_docker]: echo is frowned upon in zsh community, better use print or printf. In your case, just use:

```
ctrl_keys=(${:-ctrl-{{a..z},{@,'[','\\','^',']',_}}})
```

**Comment by vincentbernat** [comment]
Added a tiny nitpick inline. Otherwise, LGTM.

**Comment by thaJeztah** [comment]
Thanks @vincentbernat!

**Comment by vdemeester** [comment]
Once @vincentbernat comment taken care of, LGTM 🐸

**Comment by sdurrheimer** [comment]
Good to go.

**Pull request merged**